cl2_test.go

```diff
@@ -0,0 +1,230 @@
+package quamina
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+)
+
+var (
+	cl2Lock      sync.Mutex
+	cl2Lines     [][]byte
+	cl2LineCount int
+)
+
+/* This test adopted, with thanks, from aws/event-ruler */
+
+func getCL2Lines(t *testing.T) [][]byte {
+	t.Helper()
+
+	cl2Lock.Lock()
+	defer cl2Lock.Unlock()
+	if cl2Lines != nil {
+		return cl2Lines
+	}
+	file, err := os.Open("testdata/citylots2.json.gz")
+	if err != nil {
+		t.Error("Can't open citlots.jlines.gz: " + err.Error())
```

Comment:
nit: those should be `t.Fatal` to immediately abort this test (same below)

Reply:
Fixed